spec.emu

```diff
@@ -63,23 +63,28 @@ contributors: Google, Ecma International
   <emu-clause id="sec-intl-displaynames-abstracts">
     <h1>Abstract Operations for DisplayNames Objects</h1>
 
-    <emu-clause id="sec-isvalidcodefordisplaynames" aoid="IsValidCodeForDisplayNames">
-      <h1>IsValidCodeForDisplayNames ( _type_,  _code_ )</h1>
+    <emu-clause id="sec-canonicalcodefordisplaynames" aoid="CanonicalCodeForDisplayNames">
+      <h1>CanonicalCodeForDisplayNames ( _type_,  _code_ )</h1>
       <p>
-        The IsValidCodeForDisplayNames abstract operation is called with arguments _type_, and _code_. It verifies that the _code_ argument represents a well-formed code according to the _type_ argument. The algorithm refers to <a href="https://www.unicode.org/reports/tr35/#Identifiers">UTS 35's Unicode Language and Locale Identifiers grammar</a>. The following steps are taken:
+        The CanonicalCodeForDisplayNames abstract operation is called with arguments _type_, and _code_. It verifies that the _code_ argument represents a well-formed code according to the _type_ argument and returns the case-regularized form of the _code_. The algorithm refers to <a href="https://www.unicode.org/reports/tr35/#Identifiers">UTS 35's Unicode Language and Locale Identifiers grammar</a>. The following steps are taken:
       </p>
       <emu-alg>
         1. If _type_ is `"language"`, then
-          1. If _code_ matches the `unicode_language_id` production, return *true*.
-          1. Return *false*.
+          1. If _code_ does not matches the `unicode_language_id` production, throw a *RangeError* exceptipon.
+          1. Let _code_ be the result of mapping _code_ to lower case as described in <emu-xref href="#sec-case-sensitivity-and-case-mapping"></emu-xref>.
```

Comment:
`unicode_language_id` contains more than just `unicode_language_subtag`. Maybe it's best just to refer to <https://unicode.org/reports/tr35/tr35.html#Canonical_Unicode_Locale_Identifiers> here?

Reply:
opps. forgot about that. Let me reopen #70